### DISCUSSION TITLE
Error (Xcode): Signing for "OneSignalNotificationServiceExtension" requires a development team. Select a development team in the Signing & Capabilities editor.

### DISCUSSION THREAD

**Discussion by emappstore777** [codemagic-ci-cd/codemagic-docs#2220] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Editing podfile, added some recommended scripts even through documentation
Tried manual signing through .p12 certificate

### Expected results

I added One Signal for my project. I have a developer account. But sadly I can not build my project through CodeMagic. Please assist.

### Actual results

Error while building: Error (Xcode): Signing for "OneSignalNotificationServiceExtension" requires a development team. Select a development team in the Signing & Capabilities editor.

### Build id (optional)

656242d99460747a15a3dbca

**Comment by dtrdic** [comment]
Hey @emappstore777, the error message suggests that the profile cannot be found during the code signing process. It is either not created yet or has not been uploaded.
Checking at the logs I can see that you have not uploaded the provisioning profile with OneSignalNotificationServiceExtension under the iOS code signing section. Therefore when setting up code signing settings with the given profiles, you are getting the following error:

`Did not find provisioning profile matching bundle identifier "com.lennoyoga.ly.OneSignalNotificationServiceExtension" for target "OneSignalNotificationServiceExtension" [Release] from project "Runner"`

In order to fix it, please upload the profile with that bundle identifier under code signing section first, but beforehand do not forget to create one in your Apple Developer Account

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion due to a lack of activity. You can open the discussion again by replying to this message or opening a new discussion.
